src/context/analysisResultStateSlice.ts

```diff
@@ -386,6 +403,65 @@ export const requestAndStoreAnalysis = createAsyncThunk<
   );
 });
 
+export const requestAndStorePolygonAnalysis = createAsyncThunk<
+  PolygonAnalysisResult,
+  PolygonAnalysisDispatchParams,
+  CreateAsyncThunkTypes
+>('analysisResultState/requestAndStorePolygonAnalysis', async (params, api) => {
+  const { adminLevel, hazardLayer, startDate, endDate, extent } = params;
+
+  const zones = await fetchAdminLayerGeoJSON(adminLevel);
+
+  const classes = await fetchWMSLayerAsGeoJSON({
+    lyr: hazardLayer,
+    startDate,
+    endDate,
+  });
+
+  const result = calculate({
+    zones,
+    zone_properties: [getAdminNameProperty(adminLevel)],
+    classes,
+    class_properties: hazardLayer?.zonal?.class_properties, // eslint-disable-line camelcase
+    preserve_features: true,
+  });
+
+  const tableColumns = result.table.columns;
+
+  const tableRows = result.table.rows.map((row: any, i: number) => {
+    // eslint-disable-next-line no-param-reassign
+    row['stat:area'] = Math.round(
+      convertArea(row['stat:area'], 'meters', 'kilometers'),
+    );
+
+    // preformat null values as "null"
+    // so they display in the table
+    Object.keys(row).forEach((key: string) => {
+      const v: any = row[key];
+      if (v === null) {
+        row[key] = 'null'; // eslint-disable-line no-param-reassign
+      }
+    });
+
+    // create key for AnalysisTable
+    // eslint-disable-next-line no-param-reassign
+    row.name = i;
+
+    return row;
```

Comment:
this is fixed.

fwiw, performance and memory usage is exactly why I chose a mutable approach, but I supposed I should have added a comment about that.  I did test locally and it doesn't seem that the performance is much different between the two approaches in practice, probably because the data sizes are small enough.  Happy to take this up again if we find performance to be an issue.